lib/pathnames.lisp

```diff
@@ -281,16 +281,17 @@
 			       :host (pathname-host pathname)
 			       :device (pathname-device pathname)
 			       :directory (subseq dir 0 i))))
-		 (unless (probe-file newpath)
+		 (unless (directoryp newpath)
 		   (let ((namestring (native-translated-namestring newpath)))
 		     (when verbose
 		       (format *standard-output* "~&Creating directory: ~A~%"
 			       namestring))
 		     (%mkdir namestring mode)
```

Comment:
I'm thinking the way to go here is to check the result of `%mkdir` and (as the function `ccl:create-directory` does earlier in the file on line 251) and call `signal-file-error` if needed.

Reply:
Done, by having CCL:CREATE-DIRECTORY and ENSURE-DIRECTORIES-EXIST call CREATE-PARENT-DIRECTORIES which combines code from both.